src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
+            {
+                try
+                {
+                    object result = CachedExpressionCompiler.Process(expression)((TModel)container);
+                    if (expression.ReturnType.GetTypeInfo().IsEnum)
+                    {
+                        var enumValue = metadata.EnumNamesAndValues[result.ToString()];
```

Comment:
1. I don't get what this line is for.
2. The indexer will blow up if `metadata.ModelType == typeof(StringComparison)` and `result` isn't between `CurrentCulture` (`0`) and `OrdinalIgnoreCase` (`5`) e.g. `(StringComparison)23`. That's completely valid but unsupported here.
3. Further `result` can easily be `null` when `metadata.ModelType` is a `Nullable<TEnum>`. The indexer will blow up then too.